src/js/api/config.js

```diff
@@ -161,11 +161,21 @@ define([
             config.playlist = configPlaylist.playlist;
         }
 
+        if (_.size(config.playlist)) {
+            config.playlist[0].preload = normalizePreload(config);
+        }
+
         config.qualityLabels = config.qualityLabels || config.hlslabels;
 
         return config;
     };
 
+    function normalizePreload(config) {
+        const preloadValues = ['none', 'metadata', 'auto'];
+        const preload = config.playlist[0].preload || config.preload;
+
+        return _.contains(preloadValues, preload) ? preload : 'metadata';
+    }
```

Comment:
This won't apply to feed loaded playlists. This kind of logic should be done in the controller and playlist/ playlist, item and source modules.